lib/src/lib.rs

```diff
@@ -389,15 +387,16 @@ impl<'a> ChunkAccessor for DefaultChunkAccessor<'a> {
             data
         };
 
-        let mut sha256 = sha2::Sha256::new();
+        let mut sha256 = Sha256::default();
         sha256.input(&data);
-        let mut sha256_digest = vec![0u8; DIGEST_SIZE];
-        sha256.result(&mut sha256_digest);
-        if sha256_digest != digest {
+        let mut vec_result = vec![0u8; DIGEST_SIZE];
+        vec_result.copy_from_slice(&sha256.result());
```

Comment:
`default()` is the method of the standard `Default` trait implemented for hashes which do not require anything for initialization, so signature can't be changed. I think adding methods to support usage of external buffers will noticeably worsen API ergonomics of hash crates. (note that all hashes in RustCrypto/hashes work through traits described in the `digest` crate) Also I don't think that it will even help with removing copies, as hash implementations keep state internally and do not rely on external buffer for that.

I think it's worth to check if you really need `Vec` for digest results or not. But of couse we can leave it's as is too. :)